clicommand/agent_start.go

```diff
@@ -240,6 +247,11 @@ var AgentStartCommand = cli.Command{
 			Usage:  "Don't allow this agent to load plugins",
 			EnvVar: "BUILDKITE_NO_PLUGINS",
 		},
+		cli.BoolFlag{
+			Name:   "no-git-submodules",
+			Usage:  "Don't automatically checkout git submodules",
+			EnvVar: "BUILDKITE_NO_GIT_SUBMODULES,BUILDKITE_DISABLE_GIT_SUBMODULES",
```

Comment:
Wouldn't `BUILDKITE_DISABLE_GIT_SUBMODULES` flow through to the bootstrap process anyway?

Reply:
It would, but nothing there would read it, the bootstrap is looking for `BUILDKITE_GIT_SUBMODULES`. You make a good point though in that previously that setting was set in hooks and would be expected to be read in the checkout hook.

Reply:
I guess this just exposes a config setting for disabling git submodules, but it's probably a breaking change from the previous behaviour.